docs/src/design/multiset.md

```diff
@@ -1,56 +1,158 @@
+
+$$lookup = \alpha_0 + \alpha_1 \cdot op_{example} + \alpha_2 \cdot i_0 + \alpha_3 \cdot i_1 + ... + \alpha_{n+2} \cdot i_n + \alpha_{n+3} \cdot o_0 + \alpha_{n+4} \cdot o_1 + ... + \alpha_{n + 2 + m} \cdot o_m$$
+
+The constraint for sending this to the bus as a request would be:
+
+$$b' \cdot lookup = b$$
+
+The constraint for sending this to the bus as a response would be:
+
+$$b' = b \cdot lookup$$
+
+However, these constraints must be combined, since it's possible that requests and responses both occur during the same cycle. 
+
+To combine them, let $u_{lookup}$ be the request value and let $v_{lookup}$ be the response value. These values are both computed the same way as shown above, but the data sources are different, since the input/output values used to compute $u_{lookup}$ come from the trace of the component that's "offloading" the computation, while the input/output values used to compute $v_{lookup}$ come from the trace of the specialized component.
+
+The final constraint can be expressed as:
+
+$$b' \cdot u_{lookup} = b \cdot v_{lookup}$$
+
+### Communication buses in Miden VM
+
+In Miden VM, the specialized components are implemented as dedicated segments of the execution trace, which include the range checker and the 3 chiplets in the Chiplets module (the hash chiplet, bitwise chiplet, and memory chiplet).
+
+Miden VM currently uses 2 buses to communicate with these components:
+
+- The chiplets bus [$b_{chip}$](./chiplets/main.md#chiplet-bus), which communicates with all of the chiplets (Hash, Bitwise, and Memory).
+- The range checker bus [$b_{range}$](./range.md#communication-bus).
+
+#### The chiplet bus
+The Chiplet bus $b_{chip}$ enables offloading bitwise, memory, and hash operations to the respective chiplet.
+
+Requests are sent to the chiplets bus by the following components:
+
+- The stack sends requests for [bitwise](./stack/u32_ops.md#u32and), [memory](./stack/io_ops.md#memory-access-operations), and [cryptographic hash operations](./stack/crypto_ops.md).
+- The decoder sends requests for [hash operations](./decoder/main.md#program-block-hashing) for program block hashing.
+
+Responses are provided by the [hash](./chiplets/hasher.md#hash-chiplet-bus-constraints), [bitwise](./chiplets/bitwise.md#bitwise-chiplet-bus-constraints), and [memory](./chiplets/memory.md#memory-row-value) chiplets.
+
+#### The range checker bus
+The range checker bus $b_{range}$ enables offloading 16-bit range checks to the Range Checker.
+
+Requests are sent to the range checker bus by the following components:
+- The Stack sends requests for 16-bit range checks during some [`u32` operations](./stack/u32_ops.md#range-checks).
+- The [Memory chiplet](./chiplets/memory.md) sends requests for 16-bit range checks against the values in the $d_0$ and $d_1$ trace columns to enforce internal consistency.
+
+Responses are provided by the [range checker](./range.md#communication-bus).
```

Comment:
Do we need these two sections? It seems like the content here is already covered in the relevant sections of range checker and chiplet busses (linked above). I guess it doesn't hurt to have them here too, but then we need to make sure we keep this updated if/when things change.

Reply:
I actually wrote it here first and then went back and edited the other sections, so maybe we don't need it any more. I just wanted to group all the constraints in one place & since it's not practical to write them all out together I figured we could at least provide all the links together. 

I would probably keep it, but it's a good point about updates, so I could go either way. If you have a preference, let me know & I'll do that.

Reply:
Let's keep it.

Reply:
Ah - right before you responded I tried something and I think it's better... It merges this info into the respective docs to make sure the descriptions are well-aligned & then removes it from here so it'll be easier to maintain. I'll submit the PR that way & if you don't like it as much I can roll it back easily.